tendrl/gluster_integration/objects/volume/atoms/delete.py

```diff
@@ -24,6 +24,6 @@ def run(self, parameters):
             ]
         )
         etcd_client = parameters['etcd_client']
-        vol_key = "clusters/%s/Volumes/%s/deleted" % (cluster_id, vol_id)
-        etcd_client.write(vol_key, "True")
+        vol_key = "clusters/%s/Volumes/%s" % (cluster_id, vol_id)
+        etcd_client.delete(vol_key, recursive=True)
```

Comment:
Marking the volume as deleted = True was a stop gap arrangement. Final deletion of the volume entry from etcd is done here.
Will update the test case for this.